cookbooks/aws-parallelcluster-scheduler-plugin/files/default/clusterstatusmgtd/clusterstatusmgtd.py

```diff
@@ -0,0 +1,435 @@
+            "--override-runlist aws-parallelcluster::update_computefleet_status"
+        )
+        _run_command(cmd, log_file=cinc_log_file)
+
+    def _update_status(self, request_status, in_progress_status, final_status):
+        if self._compute_fleet_status == request_status:
+            self._update_compute_fleet_status(in_progress_status)
+
+        self._call_update_event()
+        if self._compute_fleet_status == in_progress_status:
+            self._update_compute_fleet_status(final_status)
+
+    @log_exception(log, "handling compute fleet status transitions", catch_exception=Exception, raise_on_error=False)
+    def manage_cluster_status(self):
+        """Manage cluster status."""
+        # Initialization
+        log.info("Managing cluster...")
+        self._current_time = datetime.now(tz=timezone.utc)
+        self._compute_fleet_status = self._get_compute_fleet_status(fallback=self._compute_fleet_status)
+        log.info(f"Current compute fleet status: {self._compute_fleet_status}")
+        try:
+            if ComputeFleetStatus.is_stop_in_progress(self._compute_fleet_status):
+                self._update_status(
+                    ComputeFleetStatus.STOP_REQUESTED, ComputeFleetStatus.STOPPING, ComputeFleetStatus.STOPPED
+                )
+            elif ComputeFleetStatus.is_start_in_progress(self._compute_fleet_status):
+                self._update_status(
+                    ComputeFleetStatus.START_REQUESTED, ComputeFleetStatus.STARTING, ComputeFleetStatus.RUNNING
+                )
+        except ComputeFleetStatusManager.ConditionalStatusUpdateFailedError:
+            log.warning(
+                f"Cluster status was updated while handling a transition from {self._compute_fleet_status}. "
+                "Status transition will be retried at the next iteration",
+            )
+
+
+def _run_clusterstatusmgtd(config_file):
+    config = ClusterstatusmgtdConfig(config_file)
+    cluster_status_manager = ClusterStatusManager(config=config)
+    while True:
+        # Get loop start time
+        start_time = datetime.now(tz=timezone.utc)
+        # Get program config
+        try:
+            config = ClusterstatusmgtdConfig(config_file)
+            cluster_status_manager.set_config(config)
+        except Exception as e:
+            log.warning(
+                f"Unable to reload daemon config from {config_file}, using previous one.\nException: {e}",
+            )
+        # Configure root logger
+        try:
+            fileConfig(config.logging_config, disable_existing_loggers=False)
+        except Exception as e:
+            log.warning(
+                f"Unable to configure logging from {config.logging_config}, "
+                f"using default logging settings.\nException: {e}",
+            )
+        # Manage cluster
+        cluster_status_manager.manage_cluster_status()
+        _sleep_remaining_loop_time(config.loop_time, start_time)
+
+
+@retry(wait_fixed=_seconds(LOOP_TIME))
```

Comment:
optional: shall we implement this ourselves so that we drop the extra third party dependency?